new-tutorial/L4_message_passing.py

```diff
@@ -0,0 +1,386 @@
+"""
+Writing GNN Modules for Stochastic GNN Training
+===============================================
+
+All GNN modules DGL provides support stochastic GNN training. This
+tutorial teaches you how to write your own graph neural network module
+for stochastic GNN training. It assumes that
+
+1. You know :doc:`how to write GNN modules for full graph
+   training <3_message_passing>`.
+2. You know :doc:`how stochastic GNN training pipeline
+   works <L1_large_node_classification>`.
+
+"""
+
+import dgl
+import torch
+import numpy as np
+from ogb.nodeproppred import DglNodePropPredDataset
+
+dataset = DglNodePropPredDataset('ogbn-products')
+
+graph, node_labels = dataset[0]
+idx_split = dataset.get_idx_split()
+train_nids = idx_split['train']
+node_features = graph.ndata['feat']
+
+sampler = dgl.dataloading.MultiLayerNeighborSampler([4, 4])
+train_dataloader = dgl.dataloading.NodeDataLoader(
+    graph, train_nids, sampler,
+    batch_size=1024,
+    shuffle=True,
+    drop_last=False,
+    num_workers=0
+)
+
+input_nodes, output_nodes, bipartites = next(iter(train_dataloader))
+
+
+######################################################################
+# DGL Bipartite Graph Introduction
```

Comment:
What exactly is the Bipartite Graph?

Reply:
I don't know how to introduce it without introducing heterogeneous graphs.  We also did not introduce what is a bipartite graph precisely is in overview.

I changed it to `you have seen the concept *bipartite graph*, where nodes are divided into two parts.`